Backend/app/rider-platform/rider-app/Main/src/API/UI/HotSpot.hs

```diff
@@ -94,42 +102,54 @@ getHotspot Maps.LatLong {..} merchantId = do
 
 -- Function to find all neighboring geohashes at the same precision
 
-geoHashLengthToLatLongCentroidalDistance :: Int -> Double
-geoHashLengthToLatLongCentroidalDistance geoHashLength = case geoHashLength of
-  1 -> 5009000
-  2 -> 1252300
-  3 -> 156500
-  4 -> 39100
-  5 -> 4900
-  6 -> 1200
-  7 -> 152
-  8 -> 38
-  9 -> 5
-  10 -> 1
-  11 -> 0.14
-  12 -> 0.03
-  _ -> 0
-
-neighboringGeohashes :: String -> [String]
-neighboringGeohashes geohash = do
-  let mbCentroid :: Maybe (Double, Double) = DG.decode geohash
+geoHashCellHeight :: Int -> Double
+geoHashCellHeight precision =
+  case precision of
+    1 -> 5009000.0
+    2 -> 626150.0
+    3 -> 156500.0
+    4 -> 19550.0
+    5 -> 4890.0
+    6 -> 610.0
+    7 -> 153.0
+    8 -> 19.1
+    9 -> 4.77
+    10 -> 0.596
+    11 -> 0.149
+    12 -> 0.0186
+    _ -> 0.0
+
+neighboringGeohashes :: String -> Double -> Double -> [String]
+neighboringGeohashes geohash distance distanceThreshold = do
+  let currentGeoHash :: Maybe (Double, Double) = DG.decode geohash
   let precision = length geohash
-  let distance = geoHashLengthToLatLongCentroidalDistance precision
-  case mbCentroid of
+  case currentGeoHash of
     Just (lat, lon) -> do
-      let neighbours = getLatlongsAtDistance (Maps.LatLong lat lon) distance
-      let geoHashes = mapMaybe (\neighbour -> DG.encode precision (neighbour.lat, neighbour.lon)) neighbours
-      geoHashes
+      let neighbours = getLatlongsAtDistance (Maps.LatLong lat lon) distance distanceThreshold
+      let geoHashes = mapMaybe (\neighbour -> DG.encode precision (neighbour.lat, neighbour.lon)) (Maps.LatLong lat lon : neighbours)
+      Dl.nub geoHashes
     Nothing -> []
 
-getLatlongsAtDistance :: Maps.LatLong -> Double -> [Maps.LatLong]
-getLatlongsAtDistance origin distance =
+getLatlongsAtDistance :: Maps.LatLong -> Double -> Double -> [Maps.LatLong]
+getLatlongsAtDistance origin distance distanceThreshold =
   let bearings = [0, 45, 90, 135, 180, 225, 270, 315]
       toRadians deg = deg * pi / 180
       toDegrees rad = rad * 180 / pi
-      originRadians = Maps.LatLong (toRadians $ origin.lat) (toRadians $ origin.lon)
-      latLongs = [getPointAtDistance originRadians distance (toRadians bearing) | bearing <- bearings]
```

Comment:
you can do the below here itself:

```
latLongs = [getPointAtDistance originRadians distance' (toRadians bearing) | bearing <- bearings, distance' <- [0, distanceThreshold .. distance]]
```

Reply:
@piyushKumar-1 I removed below `getPointAtDistance` function and merge with this one with some modification as there was some issue with logic while fetching neighbour geohashes

Reply:
Are we using that function somewhere else also ? If not can you move the modification to that function itself ?

Reply:
done, also bearing should be in outer loop